### DISCUSSION TITLE
Is `rnvim_home` correct on Windows?

### DISCUSSION THREAD

**Discussion by jalvesaq** [R-nvim/R.nvim#186] [Q&A] [answered]
When converting VimScript to Lua in [zotcite](https://github.com/jalvesaq/zotcite), the command `vim.fn.expand("<script>:h:h")` (used in `R.nvim`) didn't work as expected and the solution that I found was to extract the correct path from the `vim.o.runtimepath`. So, my question is: Does `vim.fn.expand("<script>:h:h")` work on Windows as it does in `R.nvim` on Linux? The command to allow one to answer the question is:

```vim
RConfigShow rnvim_home
```

The code used in `zotcite` is:

```lua
    local p = vim.o.runtimepath
    p = p:gsub(".*,(.*zotcite.*)", "%1")
    p = p:gsub(",.*", "")
    config.zotcite_home = p .. "/python3"
```

Even if the value of `rnvim_home` is correct on Windows, perhaps it would be safer to use the algorithm above in `R.nvim` too...

**Comment by jalvesaq** [comment]
Is this a better option?

```diff
diff --git a/lua/r/config.lua b/lua/r/config.lua
index e4704a6..72579be 100644
--- a/lua/r/config.lua
+++ b/lua/r/config.lua
@@ -292,7 +292,7 @@ local do_common_global = function()
     config.is_darwin = config.uname == "Darwin"
 
     -- config.rnvim_home should be the directory where the plugin files are.
-    config.rnvim_home = vim.fn.expand("<script>:h:h")
+    config.rnvim_home = debug.getinfo(1).short_src:gsub("/lua.*", "")
 
     -- config.uservimfiles must be a writable directory. It will be config.rnvim_home
     -- unless it's not writable. Then it wil be ~/.vim or ~/vimfiles.
```

@akthe-at, could you check this, please?

**Reply by akthe-at** [reply]
If I use `debug.getinfo(1).short_src:gsub("/lua.*", "")` it creates an odd shortened path of "...010/AppData/Local/nvim-data/lazy/R.nvim"
 and errors out when trying to read an R file.

For reference my full path would be C:\Users\ARK010\AppData\Local\nvim-data\lazy\R.nvim
So you can see where its cutting off at with the ...

Did I understand what you were asking me to check correctly or did you want me to install `zotcite` and check there?

**Comment by jalvesaq** [comment]
It was just that. No need to install zotcite. The path, although odd, seems to be correct. Is it the same of `vim.fn.expand("<script>:h:h")`?

**Reply by akthe-at** [reply, marked as answer]
It seems like a broken path to me:

```lua
    -- config.rnvim_home should be the directory where the plugin files are.
    config.rnvim_home = debug.getinfo(1).short_src:gsub("/lua.*", "")
    vim.notify("test 1: " .. config.rnvim_home)

    config.rnvim_home = vim.fn.expand("<script>:h:h")
    vim.notify("test 2: " .. config.rnvim_home)
```
   Info  12:21:35 PM notify.info test 1: ...010/AppData/Local/nvim-data/lazy/R.nvim
   Info  12:21:35 PM notify.info test 2: C:\Users\ARK010\AppData\Local\nvim-data\lazy\R.nvim

**Comment by jalvesaq** [comment]
Yes, it's broken. Perhaps because we are using the field `short_src` and not `source`. Anyway, the relevant information is that `vim.fn.expand("<script>:h:h")` works which means that we don't have to change anything.

Thank you!
